src/types/auth.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Type definitions for authentication types
+ *
+ * TODO: this may move to local component files once they are correctly typed
+ */
+
+/**
+ * @typedef AuthState
```

Comment:
I want to put this in `hypothesis-app.js`, but that file is root level and comes with a mountain of typecheck errors. We'll get there...

Reply:
The `authStateFromProfile` function in `hypothesis-app.js` could also be moved to the `session` module in the store as a selector in future.

Can I suggest moving this type to `user-menu.js` in the meantime and then just importing it from there in other components?